sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLoggingPolicy.java

```diff
@@ -122,7 +122,7 @@ public HttpLoggingPolicy(HttpLogOptions httpLogOptions) {
 
     private void formatAllowableHeaders(Set<String> allowedHeaderNames, HttpHeaders requestResponseHeaders,
                                         ClientLogger logger) {
-        if (allowedHeaderNames != null && !allowedHeaderNames.isEmpty()) {
+        if (!allowedHeaderNames.isEmpty()) {
             StringBuilder sb = new StringBuilder();
             for (HttpHeader header : requestResponseHeaders) {
                 sb.append(header.getName()).append(":");
```

Comment:
This should be case insensitive

Reply:
updated here - https://github.com/Azure/azure-sdk-for-java/pull/5893/files#diff-1bf1978289f15f56cf8db11581def1a9R129